sympy/functions/elementary/tests/test_complexes.py

```diff
@@ -892,3 +892,18 @@ def test_issue_14238():
     # doesn't cause recursion error
     r = Symbol('r', real=True)
     assert Abs(r + Piecewise((0, r > 0), (1 - r, True)))
+
+def test_zero_assumptions():
+    nr = Symbol('nonreal', real=False)
+    ni = Symbol('nonimaginary', imaginary=False)
+    # imaginary implies not zero
```

Comment:
I don't find this comment useful here

Reply:
This is something we've wanted to change for some time (see https://github.com/sympy/sympy/issues/7649 and linked discussions). I think it's quite unintuitive that `imaginary` doesn't include zero, so I noted it here to indicate why I am testing both cases.